source/main/gui/panels/GUI_TopMenubar.cpp

```diff
@@ -42,13 +43,65 @@
 #include "SkyManager.h"
 #include "Terrain.h"
 #include "Water.h"
+#include "ScriptEngine.h"
+#include "Console.h"
+#include "ContentManager.h"
 
 #include <algorithm>
 #include <fmt/format.h>
+#include <rapidjson/document.h>
+
+#ifdef USE_CURL
+#   include <curl/curl.h>
+#   include <curl/easy.h>
+#endif //USE_CURL
+
+#if defined(_MSC_VER) && defined(GetObject) // This MS Windows macro from <wingdi.h> (Windows Kit 8.1) clashes with RapidJSON
+#   undef GetObject
+#endif
 
 using namespace RoR;
 using namespace GUI;
 
+#if defined(USE_CURL)
+
+static size_t CurlWriteFunc(void *ptr, size_t size, size_t nmemb, std::string* data)
```

Comment:
all this stuff needs to be separated to it's own GUI class, like VehicleAiPanel:
* curlWriteFunc
* GetJson
* everything under `case TopMenu::TOPMENU_AI:` should be `VehicleAiPanel::Draw()`
* `TopMenubar::Refresh`
* `TopMenubar::GetPresets`